networkx/drawing/nx_pylab.py

```diff
@@ -1145,73 +1251,197 @@ def draw_networkx_edge_labels(
+            cx = (x1 + x2) / 2 + rad * (y2 - y1)  # * aspect_ratio
+            cy = (y1 + y2) / 2 - rad * (x2 - x1)  # / aspect_ratio
+
+            # Text position at a proportion t along the line in display coords
+            # default is 0.5 so text appears at the halfway point
+            x = (1 - t) * ((1 - t) * x1 + t * cx) + t * ((1 - t) * cx + t * x2)
+            y = (1 - t) * ((1 - t) * y1 + t * cy) + t * ((1 - t) * cy + t * y2)
+            x, y = self.ax.transData.inverted().transform((x, y))
+            if self.labels_horizontal:
+                # Horizontal text labels
+                angle = 0
+            else:
+                # Labels parallel to curve
+                change_x = 2 * (1 - t) * (cx - x1) + 2 * t * (x2 - cx)
+                change_y = 2 * (1 - t) * (cy - y1) + 2 * t * (y2 - cy)
+                angle = (np.arctan2(change_y, change_x) / (2 * np.pi)) * 360
+                # Text is "right way up"
+                if angle > 90:
+                    angle -= 180
+                if angle < -90:
+                    angle += 180
+            return x, y, angle
+
+        def draw(self, renderer):
+            # recalculate the text position and angle
+            self._update_text_pos_angle(self.arrow)
+            self.set_position((self.x, self.y))
+            self.set_rotation(self.angle)
+            # redraw text
+            mpl.text.Text.draw(self, renderer)
+
+    # use default box of white with white border
+    if bbox is None:
+        bbox = {"boxstyle": "round", "ec": (1.0, 1.0, 1.0), "fc": (1.0, 1.0, 1.0)}
+
+    if isinstance(connectionstyle, str):
+        connectionstyle = [connectionstyle]
+    elif np.iterable(connectionstyle):
+        connectionstyle = list(connectionstyle)
+    else:
+        raise nx.NetworkXError(
+            "draw_networkx_edges arg `connectionstyle` must be"
+            "string or iterable of strings"
+        )
+
     if ax is None:
         ax = plt.gca()
+
     if edge_labels is None:
-        labels = {(u, v): d for u, v, d in G.edges(data=True)}
-    else:
-        labels = edge_labels
-        # Informative exception for multiedges
-        try:
-            (u, v) = next(iter(labels))  # ensures no edge key provided
-        except ValueError as err:
+        kwds = {}
+        if isinstance(G, nx.MultiGraph):
+            kwds = {"keys": True}
+        edge_labels = {tuple(edge): d for *edge, d in G.edges(data=True, **kwds)}
+    # NOTHING TO PLOT
+    if not edge_labels:
+        return {}
+    edgelist, labels = zip(*edge_labels.items())
+
+    if nodelist is None:
+        nodelist = list(G.nodes())
+
+    # set edge positions
+    edge_pos = np.asarray([(pos[e[0]], pos[e[1]]) for e in edgelist])
+
+    if len(edgelist[0]) == 3:
+        # MultiGraph input
+        edge_keys = [e[2] for e in edgelist]
+        max_multi_edges = max([e[2] for e in G.edges(keys=True)]) + 1
+        if len(connectionstyle) < max_multi_edges:
```

Comment:
The usual case is when `len(connectionstyle) is 1`.  That just means all edges have the same connectionstyle. So this check is leading to false exceptions being raised.

also, I don’t think we should check for the max edge key value. We should be counting how many edges. The edge keys are not always going to be consecutive integers (though that is the default).  I don’t think we should be counting how many edges in the whole graph either… just the ones in the edgelist.  Counting edges might be expensive.  Maybe there’s another way to finesse this issue — like not checking, but raising when it is trying to get a connectionstyle and runs out of them.

Reply:
It is not what is happening now. It graph is Graph or DiGraph, then it will always take the connectionstyle[0], otherwise, it will be accessing with edge key. As in connectionstyle[edge_key].

Yeah, that was my initial idea to catch the error later. Thus why I left it without this check.

If it was to be left like it is, then the maximum index of edge_key is a sensible check. I am open to ideas for alternatives. If it was to be taken as a sequence, then some sort of edge counting of each pair would need to be introduced.

I am open to ideas, however I feel that alternatives might be much more complex than the current approach.